packages/swayze/test/fill_selections_test.dart

```diff
@@ -0,0 +1,374 @@
+
+      if (goldenNameOngoing != null) {
+        await expectLater(
+          find.byType(TestSwayzeVictim),
+          matchesGoldenFile('goldens/$goldenNameOngoing.png'),
+        );
+      }
+
+      await gesture.up(
+        timeStamp: const Duration(seconds: 8),
+      );
+      await gesture.removePointer();
+
+      await tester.pumpAndSettle(
+        const Duration(seconds: 10),
+      );
+
+      if (goldenNameEnd != null) {
+        await expectLater(
+          find.byType(TestSwayzeVictim),
+          matchesGoldenFile('goldens/$goldenNameEnd.png'),
+        );
+      }
+
+      expect(
+        fillSource,
+        startRange,
+        reason: 'Should call FillIntoTargetIntent with correct source range.',
+      );
+
+      expect(
+        fillTarget,
+        targetRange,
+        reason: 'Should call FillIntoTargetIntent with correct target range.',
+      );
+    },
+  );
+}
+
+void _testFillUnknown() {
+  testWidgets(
+    'Fill Unknown Selection',
+    (WidgetTester tester) async {
+      Range2D? fillSource;
+
+      await _createTable(
+        tester,
+        actions: {
+          FillIntoUnknownIntent: CallbackAction<FillIntoUnknownIntent>(
+            onInvoke: (intent) {
+              fillSource = Range2D.fromPoints(
+                intent.source.leftTop,
+                intent.source.rightBottom,
+              );
+
+              return null;
+            },
+          ),
+        },
+      );
+
+      const rangeStart = IntVector2(1, 1);
+      const rangeEnd = IntVector2(1, 3);
+
+      final startOffset = getCellOffset(
+        tester,
+        column: rangeStart.dx,
+        row: rangeStart.dy,
+      );
+      final endOffset = getCellOffset(
+        tester,
+        column: rangeEnd.dx,
+        row: rangeEnd.dy,
+      );
+
+      await tester.pumpAndSettle();
+
+      final gesture = await tester.startGesture(
+        startOffset,
+        kind: PointerDeviceKind.mouse,
+      );
+
+      // Move far away and then move back so that the drag registers on the test
+      await gesture.moveTo(
+        const Offset(1000.0, 1000.0),
+        timeStamp: const Duration(seconds: 2),
+      );
+
+      await gesture.moveTo(
+        endOffset,
+        timeStamp: const Duration(seconds: 4),
+      );
+
+      await tester.pumpAndSettle(
+        const Duration(seconds: 6),
+      );
+
+      await gesture.up(
+        timeStamp: const Duration(seconds: 8),
+      );
+
+      await gesture.removePointer();
+
+      await tester.pumpAndSettle(
+        const Duration(seconds: 10),
+      );
+
+      await expectLater(
+        find.byType(TestSwayzeVictim),
+        matchesGoldenFile('goldens/fill_selection_multiple_cells_handle.png'),
+      );
+
+      final handlerOffset = endOffset.toHandlerOffset(
+        tester,
+        endOffset: getCellOffset(
+          tester,
+          column: rangeEnd.dx + 1,
+          row: rangeEnd.dy + 1,
+        ),
+      );
+
+      await tester.tapAt(handlerOffset);
```

Comment:
It would be nice to have an extension for this gesture, so the reader would read `tester.doubleTapAt` and would understand right away what is happening here.

Reply:
Done.